targets/esp-idf/io.c

```diff
@@ -0,0 +1,143 @@
+ * distributed under the License is distributed on an "AS IS" BASIS
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include <stdarg.h>
+#include <stdlib.h>
+#include <string.h>
+#include "esp_log.h"
+
+#include "jerryscript-port.h"
+#include "jerryscript-debugger.h"
+
+static const char TAG[] = "JS";
+
+static esp_log_level_t crosslog(jerry_log_level_t level) {
+	switch(level) {
+	case JERRY_LOG_LEVEL_ERROR: return ESP_LOG_ERROR;
+	case JERRY_LOG_LEVEL_WARNING: return ESP_LOG_WARN;
+	case JERRY_LOG_LEVEL_DEBUG: return ESP_LOG_DEBUG;
+	case JERRY_LOG_LEVEL_TRACE: return	ESP_LOG_VERBOSE;
+	}
+	return ESP_LOG_NONE;
+}
+
+#ifndef DISABLE_EXTRA_API
+
+/**
+ * Actual log level
+ */
+static jerry_log_level_t jerry_port_default_log_level = JERRY_LOG_LEVEL_ERROR;
+
+#define JERRY_PORT_DEFAULT_LOG_LEVEL jerry_port_default_log_level
+
+/**
+ * Get the log level
+ *
+ * @return current log level
+ *
+ * Note:
+ *      This function is only available if the port implementation library is
+ *      compiled without the DISABLE_EXTRA_API macro.
+ */
+jerry_log_level_t
+jerry_port_default_get_log_level (void)
+{
+  return jerry_port_default_log_level;
+} /* jerry_port_default_get_log_level */
+
+/**
+ * Set the log level
+ *
+ * Note:
+ *      This function is only available if the port implementation library is
+ *      compiled without the DISABLE_EXTRA_API macro.
+ */
+void
+jerry_port_default_set_log_level (jerry_log_level_t level) /**< log level */
+{
+  jerry_port_default_log_level = level;
+} /* jerry_port_default_set_log_level */
+
+#else /* DISABLE_EXTRA_API */
+#define JERRY_PORT_DEFAULT_LOG_LEVEL JERRY_LOG_LEVEL_ERROR
+#endif /* !DISABLE_EXTRA_API */
+
+/**
+ * Default implementation of jerry_port_log. Prints log message to the standard
+ * error with 'vfprintf' if message log level is less than or equal to the
+ * current log level.
+ *
+ * If debugger support is enabled, printing happens first to an in-memory buffer,
+ * which is then sent both to the standard error and to the debugger client.
+ *
+ * Note:
+ *      Changing the log level from JERRY_LOG_LEVEL_ERROR is only possible if
+ *      the port implementation library is compiled without the
+ *      DISABLE_EXTRA_API macro.
+ */
+void
+jerry_port_log (jerry_log_level_t level, /**< message log level */
+                const char *format, /**< format string */
+                ...)  /**< parameters */
+{
+  if (level <= JERRY_PORT_DEFAULT_LOG_LEVEL)
+  {
+    va_list args;
+    va_start (args, format);
+#if defined (JERRY_DEBUGGER) && (JERRY_DEBUGGER == 1)
+    int length = vsnprintf (NULL, 0, format, args);
+    va_end (args);
+    va_start (args, format);
+
+    JERRY_VLA (char, buffer, length + 1);
+    vsnprintf (buffer, (size_t) length + 1, format, args);
+
+	esp_log_write(crosslog(level), TAG, buffer);
```

Comment:
Incorrect indentation.

Reply:
I'm using tabs for 4 spaces and forgot to change my editor config